### PR TITLE
Use std::array<> in PhaseUsage

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/core/props/phaseUsageFromDeck.hpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ namespace Opm
     /// Determine the active phases
     inline PhaseUsage phaseUsage(const Phases& phases) {
         PhaseUsage pu;
-        std::fill(pu.phase_used, pu.phase_used + BlackoilPhases::MaxNumPhases + BlackoilPhases::NumCryptoPhases, 0);
+        std::fill(pu.phase_used.begin(), pu.phase_used.end(), 0);
```
Comment (bska) [inline opm/core/props/phaseUsageFromDeck.hpp:37]:
```C++
pu.phase_used.fill(0);
```

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by bska** [comment]
Build check is green.  I'll merge into master.

**Comment by blattms** [comment]
A very simple value initialization should also do the trick for array:
```
PhaseUsage pu{}; // initialized to zeros
```

**Comment by bska** [comment]
>```C++
> PhaseUsage pu{}; // initialized to zeros
> ```

It's not clear to me if that's sufficient.  The [`PhaseUsage` structure](https://github.com/OPM/opm-simulators/blob/2574634c5f5193f584d62ad6bade23e2b8a31d8a/opm/core/props/BlackoilPhases.hpp#L44) doesn't have initialisers for its members so while the above is okay for the array members, it might not be fully well defined for the others, particularly the `bool has*` members.  To be completely safe, we should amend the `PhaseUsage` structure as follows too
```C++
struct PhaseUsage : public BlackoilPhases
{
    std::array<int, MaxNumPhases + NumCryptoPhases> phase_used{};
    std::array<int, MaxNumPhases + NumCryptoPhases> phase_pos{};

    int num_phases{0};
    bool has_solvent{false};
    bool has_polymer{false};
    bool has_energy{false};
    // polymer molecular weight
    bool has_polymermw{false};
    bool has_foam{false};
    bool has_brine{false};
    bool has_zFraction{false};
};
```

**Pull request merged**